src/Helper/ConvertOrder.php

```diff
@@ -214,11 +221,28 @@ private function getTaxRatesShipments(): ?float
 
     private function getUnitPriceWithTax(OrderItem $item): int
     {
-        return (int) round($item->getUnitPrice() + ($item->getTaxTotal() / $item->getQuantity()));
+        $taxRate = $this->taxRateResolver->resolve($item->getVariant());
+
+        if ($taxRate === null) {
+            return $item->getUnitPrice();
+        }
+
+        if ($taxRate->isIncludedInPrice()) {
+            return $item->getUnitPrice();
+        }
+
+        return $item->getAdjustmentsTotalRecursively(AdjustmentInterface::TAX_ADJUSTMENT);
     }
 
-    private function getUnitDiscountAmount(OrderItem $item): int
+    private function getItemDiscountAmount(OrderItem $item): int
     {
-        return ($item->getUnitPrice() - $item->getFullDiscountedUnitPrice()) * $item->getQuantity();
+        $adjustments = [AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT, AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT, AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT];
```

Comment:
change adjustments to adjustemnsType because this is not exactly adjustments only types